docker/autoscaler/run_autoscaler.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from ray import ray_constants
+from ray._private.ray_logging import setup_component_logger
+from ray._private.services import get_node_ip_address
+from ray.autoscaler._private.monitor import Monitor
+import yaml
+
+
+AUTOSCALING_CONFIG_PATH = "/autoscaler/ray_bootstrap_config.yaml"
+AUTOSCALING_LOG_DIR = "/tmp/ray/session_latest/logs/"
+
+
+def setup_logging() -> None:
+    """Setup logging to
+    - log at INFO level to standard autoscaler log location. (logs viewable in UI)
+    - log at DEBUG level to pod stout. (logs viewable with `kubectl logs <head-pod> -c autoscaler`)
+    """
+    # Write logs at info level to monitor.log.
+    setup_component_logger(
+        logging_level=ray_constants.LOGGER_LEVEL,  # info
+        logging_format=ray_constants.LOGGER_FORMAT,
+        log_dir=AUTOSCALING_LOG_DIR,
+        filename=ray_constants.MONITOR_LOG_FILE_NAME,  # monitor.log
+        max_bytes=ray_constants.LOGGING_ROTATE_BYTES,
+        backup_count=ray_constants.LOGGING_ROTATE_BACKUP_COUNT,
+    )
+
+    # Also log at DEBUG level to stdout for debugging with `kubectl logs`.
+    root_logger = logging.getLogger("")
+    root_logger.setLevel(logging.DEBUG)
+
+    root_handler = logging.StreamHandler()
+    root_handler.setLevel(logging.DEBUG)
+    root_handler.setFormatter(logging.Formatter(ray_constants.LOGGER_FORMAT))
+
+    root_logger.addHandler(root_handler)
+
+
+if __name__ == "__main__":
+    setup_logging()
+    cluster_name = yaml.safe_load(open(AUTOSCALING_CONFIG_PATH).read())["cluster_name"]
+    head_ip = get_node_ip_address()
+    Monitor(
+        redis_address=f"{head_ip}:6379",
+        redis_password=ray_constants.REDIS_DEFAULT_PASSWORD,
```

Comment:
user can customize the redis password when they create the cluster. I think we will need to use that config here

Reply:
Eventually.
And also we'll want to support not using Redis at all when that's generally available.

These days, the Ray team's messaging is that redis password is not part of the public API. One shouldn't expect the password to guarantee security -- you should secure your network instead.

Reply:
If using a custom redis password, probably you'd want to use a k8s secret.

Reply:
Thanks for the discussion! I introduced some very light weight way for users to configure the password without rebuilding the docker images etc. by just putting the autoscaler pod startup command into the k8s pod specification instead of the docker file and introducing an argument for it, I think that should be sufficient for now and will allow people who have a custom password to use this code.

Longer term I don't think this is relevant, since there is an effort to get rid of redis as a dependency for the GCS at the moment.

https://github.com/ray-project/ray/pull/21086/commits/6243a38123f628a5dcd3ac17c690b013d21fdc59